e2e-pw/src/oss/specs/smoke-tests/saved-views.spec.ts

```diff
@@ -0,0 +1,260 @@
+import { test as base, expect } from "src/oss/fixtures";
+import { SavedViewsPom, updatedView2 } from "src/oss/poms/saved-views";
+import { getUniqueDatasetNameWithPrefix } from "src/oss/utils";
+
+const datasetName = getUniqueDatasetNameWithPrefix("smoke-quickstart");
+
+const test = base.extend<{ savedViews: SavedViewsPom }>({
+  savedViews: async ({ page }, use) => {
+    await use(new SavedViewsPom(page));
+  },
+});
+
+test.describe("saved views", () => {
+  test.beforeAll(async ({ fiftyoneLoader }) => {
+    await fiftyoneLoader.loadZooDataset("quickstart", datasetName, {
+      max_samples: 5,
+    });
+  });
+
+  test.beforeEach(async ({ page, fiftyoneLoader, savedViews }) => {
+    await fiftyoneLoader.waitUntilLoad(page, datasetName);
+
+    const hasUnsaved = await savedViews.unsavedViewCount();
+    if (!hasUnsaved) {
+      await savedViews.clearView();
+    }
+
+    await savedViews.openSelect();
+    const count = await savedViews.savedViewOptionCount();
+
+    if (count) {
+      await savedViews.clickOptionEdit();
+      await savedViews.clickDeleteBtn();
+    } else {
+      // close select
+      await savedViews.openSelect();
```

Comment:
Shot in the dark, but could we encapsulate all of this logic in the POM in a method like `resetSavedView`?